### DISCUSSION TITLE
Extracting polarization information

### DISCUSSION THREAD

**Discussion by Hayden-Higgins** [NanoComp/meep#2275] [Q&A]
I'm attempting a simulation where waves of a controlled frequency and linear polarization are launched (in the z direction) through a rectangular prism with certain parameters. I'm seeking a predictable relationship between the physical characteristics of this prism and the polarization state of the wave as it propagates out the other side. 

My thinking is that the most reasonable way to characterize these states is via the relative phase of the x and y components of the electric field, as well as their magnitudes to fully specify an elliptical polarization state. From browsing the MEEP documentation, I have a hunch that the get_eigenmode_coefficients method is the best way to extract this info, but I'm afraid my theoretical background isn't extensive enough to grasp which information will be useful to me. Do you have any suggestions for how to proceed?

Apologies for the open ended question, I'm not the most experienced in this area of physics/electrical engineering. Thanks for your time.

**Comment by smartalecH** [comment]
You are on the right track. One way to characterize the behavior you are after is by computing _scattering parameters_, which are essentially the linear transfer function of the device for each mode/polarization in the frequency domain. 

Doing so requires computing mode overlaps, like you discuss. Essentially, place a mode monitor before your object and one after. Compute the forward propagating coefficient for both monitors and evaluate the ratio. You can do this for both polarization states. The result will tell you the complex _field_ transformation that occurs at a specified wavelength point. (You can square the value to identify how much power is transmitted in each polarization). 

You can view an example within the context of guided optics here:  https://meep.readthedocs.io/en/latest/Python_Tutorials/Mode_Decomposition/

**Reply by oskooi** [reply]
To add to @smartalecH's suggestion: since it seems your problem involves planewaves in 3d, you will need to use a [`DiffractedPlanewave`](https://meep.readthedocs.io/en/latest/Python_User_Interface/#diffractedplanewave) object for the mode decomposition.
